src/auth.rs

```diff
@@ -0,0 +1,80 @@
+use honey_badger::AuthLevel;
+use std::sync::Arc;
+
+use eel::errors::Result;
+use eel::key_derivation::derive_key_pair_hex;
+use eel::MapToError;
+
+// Backwards-compatibility / Technical debt mess
+// Be ready for a little story telling...
+// Apparently, the BACKEND_WALLET_DERIVATION_PATH should be the actual key used for authorization
+// However, because they used a the derivation math "m" (master key) for that in other services
+// they figured it would be insecure to keep this key in memory all the time (which is true),
+// just for authorizing oneself towards the backend.
+// So their solution was to sign a secondary, somewhat ephemeral key pair with the master key,
+// and then use that key pair for authorization (here called BACKEND_AUTH_DERIVATION_PATH).
+// Since this code right here is using a sensible key derivation path from the get-go,
+// this scheme would not be necessary. However, the backend still expects the old scheme,
+// so that's why we just derive 2 key pairs, and follow the overhead of the old scheme.
+// For the ones who have access, follow this issue: https://getlipa.atlassian.net/browse/APP-1057
+const BACKEND_WALLET_DERIVATION_PATH: &str = "m/76738065'/0'/0";
+const BACKEND_AUTH_DERIVATION_PATH: &str = "m/76738065'/0'/2";
+
+pub struct Auth {
+    auth: Arc<honey_badger::Auth>,
+}
```

Comment:
Why do we need this new wrapper `Auth` struct? On `lib.rs`, we already have the `build_auth()` function which returns an instance of `honey_badger::Auth` and is used for the exchange rate provider.

Reply:
Between both approaches, I would go with the more compact `build_auth()` function.

Reply:
YES: fbc78398206eeceae1fd5e4903bf36e147f9a6a7